makes/binary.mk

```diff
@@ -31,7 +31,7 @@ SRCS += $(addprefix $(LOCAL_DIR), $(LOCAL_SRCS))
 HEADERS += $(addprefix $(LOCAL_DIR), $(LOCAL_HEADERS))
 
 # linking prerequisites
-OBJS.$(NAME) := $(patsubst %.c,$(PREFIX_O)%.o,$(SRCS))
+OBJS.$(NAME) := $(patsubst %,$(PREFIX_O)%.o,$(basename $(filter %.c %.cc %.cpp, $(SRCS))))
```

Comment:
Consider using `addsufix` instead of `pathsubst`.

Reply:
Resignation from `pathsubst` will force the use of `addsufix` and `addprefix` functions. Instead of this, `static-lib.mk` and `binary.mk` use `pathsubst`.